python/test/test_variable.py

```diff
@@ -48,10 +48,13 @@ def test_create_dtype(self):
         var = Variable(Coord.X, [Dim.X], (4,), dtype=np.dtype(np.int32))
         self.assertEqual(var.numpy.dtype, np.dtype(np.int32))
 
-    def test_coord_set_name_fails(self):
+    def test_coord_set_name(self):
         var = Variable(Coord.X, [Dim.X], np.arange(4))
-        with self.assertRaises(RuntimeError):
-            var.name = "data"
+        # We can set a name for coordinates, but not that operations will typically
```

Comment:
What has been the driver for named coordinates?

Reply:
- Avoids either one level of nesting for monitors or gets rid of an extra auxiliary dimensions `Dim::Monitors`, which is confusing. Provides named access without requiring a `DatasetIndex` (compare the old example https://github.com/mantidproject/dataset/blob/master/doc/design.md#examples-cpp).
- Coordinates that do not correspond to dimensions (those that are not "dimension coordinates") are more flexible. Consider storing data from a number of temperature sensors. Without names we would need `Coord::Temperature1`, `Coord::Temperature2`, ... which totally defeats the purpose. `(Coord::Temperature, "sample")` and `(Coord::Temperature, "cryostat")` are much clearer.
- Less code and special cases in the implementation?